plugins/package-managers/bazel/src/main/kotlin/StarlarkParser.kt

```diff
@@ -0,0 +1,239 @@
+            val start = pos
+
+            while (!isMultilineString() && !isEOF()) {
+                advance()
+            }
+            val str = input.substring(start, pos)
+            advance(3) // Skip closing """.
+            return Token(TokenType.STRING, str)
+        } else {
+            advance()
+            val start = pos
+            while (peek() != '"' && !isEOF()) {
+                advance()
+            }
+            val str = input.substring(start, pos)
+            advance() // Skip closing quote.
+            return Token(TokenType.STRING, str)
+        }
+    }
+
+    private fun tokenizeIdentifier(): Token {
+        val start = pos
+        while (isAlpha(peek()) || isDigit(peek()) || peek() == '_') {
+            advance()
+        }
+        val str = input.substring(start, pos)
+
+        if (str.lowercase() == "true" || str.lowercase() == "false") {
+            return Token(TokenType.BOOLEAN, str.lowercase())
+        }
+        return Token(TokenType.IDENTIFIER, str)
+    }
+
+    private fun tokenizeNumber(): Token {
+        val start = pos
+        while (isDigit(peek())) {
+            advance()
+        }
+        val number = input.substring(start, pos)
+        return Token(TokenType.NUMBER, number)
+    }
+
+    fun nextToken(): Token {
+        while (!isEOF()) {
+            when (val c = peek()) {
+                '#' -> { // Skip comment.
+                    while (peek() != '\n' && !isEOF()) {
+                        advance()
+                    }
+                }
+                ' ', '\n', '\r', '\t' -> advance()
+                '=' -> {
+                    advance()
+                    return Token(TokenType.EQUALS, "=")
+                }
+                '(' -> {
+                    advance()
+                    return Token(TokenType.LPAREN, "(")
+                }
+                ')' -> {
+                    advance()
+                    return Token(TokenType.RPAREN, ")")
+                }
+                '[' -> {
+                    advance()
+                    return Token(TokenType.LBRACKET, "[")
+                }
+                ']' -> {
+                    advance()
+                    return Token(TokenType.RBRACKET, "]")
+                }
+                ',' -> {
+                    advance()
+                    return Token(TokenType.COMMA, ",")
+                }
+                '"' -> {
+                    return tokenizeString()
+                }
+                else -> {
+                    if (isAlpha(c)) {
+                        return tokenizeIdentifier()
+                    } else if (isDigit(c)) {
+                        return tokenizeNumber()
+                    } else {
+                        advance() // Skip anything else to avoid an endless loop.
+                    }
+                }
+            }
+        }
+        return Token(TokenType.EOF, "")
+    }
+}
+
+class Parser(input: String) {
```

Comment:
Please reduce visibility of all classes in this file to `internal` or `private` if possible.